moshi/moshi/models/tts.py

```diff
@@ -676,3 +685,124 @@ def get_prefix(self, audio_path: Path) -> torch.Tensor:
+            voices = voice
+        elif multiple_texts and multiple_voices:
+            if len(text) != len(voice):
+                raise ValueError(
+                    f"Number of texts and voices must match, got {len(text)=} != {len(voice)=}"
+                )
+            if len(text) == 0:
+                raise ValueError("Got empty list, nothing to generate")
+            texts = text
+            voices = voice
+        else:
+            assert False, "Never happens, just making the type-checker happy"
+
+        entries_batch = [self.prepare_script([t], padding_between=1) for t in texts]
+        voice_paths = [
+            Path(v) if v.endswith(".safetensors") else self.get_voice_path(v)
+            for v in voices
+        ]
+
+        trained_with_cfg_distillation = bool(self.valid_cfg_conditionings)
+        if not trained_with_cfg_distillation:
+            # Pass the CFG coef to the model directly - otherwise we set it in the conditioning below
+            self.cfg_coef = cfg_coef
+
+        # Some models allow explicit conditioning on the speaker via cross-attention,
+        # while others can be conditioned by providing a voice prefix.
+        use_speaker_conditioning = self.multi_speaker
+        if use_speaker_conditioning:
+            condition_attributes_batch = [
+                self.make_condition_attributes([vp], cfg_coef=cfg_coef if trained_with_cfg_distillation else None)
+                for vp in voice_paths
+            ]
+            prefixes = None
+            prefix_length_frames = [0] * len(voice_paths)
+        else:
+            condition_attributes_batch = []
+            prefixes = [self.get_prefix(vp) for vp in voice_paths]
+            prefix_length_frames = [prefix.shape[-1] for prefix in prefixes]
+
+        with tqdm.auto.tqdm(disable=not show_progress, desc="Generating") as pbar:
+
+            def _on_frame(frame):
+                if (frame != -1).all():
+                    pbar.update(1)
+
+            tts_result = self.generate(entries_batch, condition_attributes_batch, on_frame=_on_frame, prefixes=prefixes)
+
+        # No progress bar for Mimi decoding because it's quick
+        with self.mimi.streaming(len(entries_batch)), torch.no_grad():
+            pcms_timewise = []
+            for frame in tts_result.frames[self.delay_steps :]:
+                pcm = self.mimi.decode(frame[:, 1:, :])
+                pcm = torch.clip(pcm, -1, 1)
+                pcms_timewise.append(pcm)
+
+        # I'm not sure why but otherwise there is a little click at the beginning
+        # and removing two frames makes the largest end_steps match len(pcms_timewise)
+        pcms_timewise = pcms_timewise[2:]
+
+        pcms_batchwise = [
+            torch.concat([pcm[i, 0, :] for pcm in pcms_timewise[prefix_length_frames[i] : tts_result.end_steps[i]]])
+            for i in range(len(entries_batch))
+        ]
+        return pcms_batchwise
+
+
+def get_default_tts_model(
```

Comment:
maybe we could have a `get_public_tts_model` ? without `n_q` as a param as it only supports 16.